### DISCUSSION TITLE
Possible to get 2FA code somehow if already logged in?

### DISCUSSION THREAD

**Discussion by backslashV** [mrlt8/docker-wyze-bridge#165] [Q&A] [answered]
I forgot to back up my Google auth codes when switching phones and I am locked out of my account. Is there a way to get the 2FA code via this app if already logged in via HA?

**Comment by mrlt8** [comment, marked as answer]
Sorry to hear that.

If you were using the totp key, then you could just copy the key over to Authenticator. 

If not, you might be able to use the access_token from your auth.pickle file to try to disable 2fa on the account.

**Comment by backslashV** [comment]
Thank you for your response. Copy the code from where? I don't have it. Also, auth.pickle contents aren't readable. What do I need to do with the contents of this file?

**Reply by mrlt8** [reply]
If you were using totp key to generate the verification code it would be in `/config/wyze-bridge/totp`

You'll have to use python to unpickle the data. Might be able to use something like `python -mpickle /path/to/auth.pickle`

**Reply by backslashV** [reply]
there is no totp folder there. When I try to unpickle, I get this error: ModuleNotFoundError: No module named 'wyzecam'

**Reply by backslashV** [reply]
ok I installed the wyzecam module and I can see the contents. There are strings for access_token and refresh_token, mfa_options=None, mfa_details=None, ...
How can I use the access_token to disable 2FA? When I try to log in to the Wyze app, it gives me the option to use my recovery code which I don't have.

**Reply by csanner** [reply]
You are out of luck here.
Your best bet is to call wyze support.

If someone could recover your totp from an old code it wouldn't be much good, I'm afraid.

**Reply by mrlt8** [reply]
@backslashV already has an active session pickled, so I think it would be possible to remove mfa with a delete request to `https://auth-prod.api.wyze.com/api/v2/users/me/mfa` with the authorization headers from the auth pickle.

**Reply by backslashV** [reply]
> @backslashV already has an active session pickled, so I think it would be possible to remove mfa with a delete request to `https://auth-prod.api.wyze.com/api/v2/users/me/mfa` with the authorization headers from the auth pickle.

Can you please specify the exact command?

**Reply by mrlt8** [reply]
This is _WAY_ beyond the scope of the bridge so never really looked into it, but as I posted earlier, I believe you need to make a DELETE request to the api with the headers `authorization: your_access_token`,`phone-id: your_phone_id` along with `user-agent: wyze_ios_2.24.52` and `x-api-key` which can be found [here](https://github.com/mrlt8/wyzecam/blob/main/wyzecam/api.py#L12-L15).

**Reply by backslashV** [reply]
Thanks. I got 401 in response.

**Reply by mrlt8** [reply]
hmm maybe your session had already expired? I just tried a `curl -X DELETE`  and it worked for me.

**Reply by backslashV** [reply]
When I use curl, I don't get any errors, but 2FA is still there.

curl -X DELETE https://auth-prod.api.wyze.com/api/v2/users/me/mfa --header 'authorization: long string here' --header 'phone-id: phone id here' --header 'user-agent: wyze_ios_2.24.52' --header 'x-api-key: WMXHYf79Nr5gIlt3r0r7p9Tcw5bvs6BB4U8O8nGJ'

Also, tried with auth.pickle contents from another Linux machine I was logged into and the same happened.

Am I using it correctly?

**Comment by backslashV** [comment]
It worked. Thanks a lot @mrlt8 for your immense help. My current access_token didn't work. However, when I restored an earlier HA backup, auth.pickle got updated, and the new access_token worked.

**Reply by mrlt8** [reply]
Glad you were able to get it sorted. The bridge attempts to minimize the number of hits to the Wyze API and, once up and running, won't ping them unless an ENR expires or the bridge is attempting to pull a thumbnail from the api.
